### PR TITLE
[Refactor] refactor train.py & test.py

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmrotate/utils/setup_env.py [hunk 1]
```diff
@@ -51,3 +53,39 @@ def setup_multi_processes(cfg):
...
+        init_default_scope (bool): Whether initialize the mmrotate default scope.
+            When `init_default_scope=True`, the global default scope will be
+            set to `mmrotate`, anmmrotate all registries will build modules from mmrotate's
+            registry node. To understand more about the registry, please refer
+            to https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/registry.md
+            Defaults to True.
+    """  # noqa
+    # TODO
+    import mmrotate.datasets  # noqa: F401,F403
+    # import mmrotate.engine  # noqa: F401,F403
```
Comment (jbwang1997) [inline mmrotate/utils/setup_env.py:71]: We need to import mmrotate.core to register TASK_UTILS.

File: tools/test.py [hunk 2]
```diff
@@ -94,163 +47,68 @@ def parse_args():
...
 
-    if args.eval and args.format_only:
-        raise ValueError('--eval and --format_only cannot be both specified')
 
-    if args.out is not None and not args.out.endswith(('.pkl', '.pickle')):
-        raise ValueError('The output file must be a pkl file.')
+def main():
+    args = parse_args()
+
+    # register all modules in mmdet into the registries
```
Comment (RangiLyu) [inline tools/test.py:77]:
```suggestion
    # register all modules in mmdet and mmrotate into the registries
```

File: tools/test.py [hunk 3]
```diff
@@ -94,163 +47,68 @@ def parse_args():
...
+    # do not init the default scope here because it will be init in the runner
+    register_all_modules(init_default_scope=False)
```
Comment (RangiLyu) [inline tools/test.py:79]: should also register mmdet modules

File: tools/train.py [hunk 4]
```diff
@@ -82,17 +47,16 @@ def parse_args():
 def main():
     args = parse_args()
 
+    # register all modules in mmdet into the registries
```
Comment (RangiLyu) [inline tools/train.py:50]:
```suggestion
    # register all modules in mmdet and mmrotate into the registries
```

File: tools/train.py [hunk 5]
```diff
@@ -82,17 +47,16 @@ def parse_args():
...
+    # do not init the default scope here because it will be init in the runner
+    register_all_modules(init_default_scope=False)
```
Comment (RangiLyu) [inline tools/train.py:52]: register mmdet modules

**Review by RangiLyu — APPROVED** [review]

**Comment by RangiLyu** [comment]
Can be merged after resolving conflicts.

**Pull request closed without merging**